pipeline_api/src/main/java/com/splicemachine/pipeline/context/PipelineWriteContext.java

```diff
@@ -124,14 +124,21 @@ public void result(KVPair put, WriteResult result) {
 
     @Override
     public void result(byte[] resultRowKey, WriteResult result) {
```

Comment:
I don't understand this change, if the rowKey is found it's updated with the result, otherwise all rows are updated with the result? What's the rationale behind it?

Reply:
I think previously it wasn't expected to reach this code path with a `resultRowKey` that is not part of the `result`.
Now it is possible when a failure from another inner pipeline is propagated, I thought the best way to handle it is to write the same result (in failing cascade it would be failure) to all rows rolling it back.
It kind of makes sense for the particular case of `on delete cascade` that I demonstrated to Jun in another comment, but I am not this does not break other code paths, therefore I would appreciate more feedback on this if possible.

Reply:
I think we need to map the lower level failure to the specific rows that failed at this level, and keep the same code we had here.

Reply:
Thanks Daniel, will do.